testdata-generator-common/src/main/java/io/openepcis/testdata/generator/model/AbstractEventCreationModel.java

```diff
@@ -284,7 +288,8 @@ public List<String> referencedEpcsIdentifierGenerator(
                 t -> epcList.addAll(EventModelUtil.instanceIdentifiers(t, epc.getEpcCount())));
       }
 
-      if (epc.getParentNodeId() != 0 && epc.getInheritParentCount() != null && epc.getInheritParentCount() > 0) {
+      //Add the identifiers from ParentID Count only if the inheriting event is ObjectEvent or TransformationEvent
+      if (epc.getParentNodeId() != 0 && epc.getInheritParentCount() != null && epc.getInheritParentCount() > 0 && inheritParentIdCountFlag) {
```

Comment:
`epc.getInheritParentCount() != null` - can't we make null-safe and always have inheritParentCount set to zero by default?
just asking, if == null or != null is an important condition somewhere else, just leave it as it is.